x/evm/abci.go

```diff
@@ -429,8 +431,10 @@ func handleConfirmedEvents(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus,
 			continue
 		}
 
+		handledEvents := 0
 		var event types.Event
-		for queue.DequeueUntil(&event, shouldHandleEvent) {
+		for queue.DequeueUntil(&event, shouldHandleEvent) && handledEvents < maxConfirmedEventsPerBlock {
```

Comment:
If the second check fails, we'd have already dequeued an event and would lose it.
```suggestion
		for handledEvents < maxConfirmedEventsPerBlock && queue.DequeueUntil(&event, shouldHandleEvent)  {
```

Reply:
good catch